src/AccessionTables/Table.js

```diff
@@ -0,0 +1,82 @@
+import PropTypes from 'prop-types';
+import { Field } from 'redux-form';
+import { FormattedMessage } from 'react-intl';
+
+import { EditableList } from '@folio/stripes/smart-components';
+
+import { LoadingCentered } from '../components';
+import { AccessionTable as AT } from '../API';
+import { Locations, AccessionTable } from '../data';
+import { Location, LocationSelection } from './components';
+
+
+const columnMapping = {
+  originalLocationId: <FormattedMessage id="ui-remote-storage.location.original" />,
+  finalLocationId: <FormattedMessage id="ui-remote-storage.location.final" />,
+};
+
+
+const LocationField = ({ name, configurationId, ...props }) => {
+  const { locations } = Locations.useByConfigurationId(configurationId);
+
+  return <Field name={name} component={LocationSelection} locations={locations} {...props} />;
+};
+
+LocationField.propTypes = {
+  ...Field.propTypes,
+  configurationId: PropTypes.string.isRequired,
+};
+
+
+// todo: guard editing with permissions
+export const Table = ({ configurationId }) => {
+  const { rows } = AccessionTable.useByConfigurationId(configurationId);
+  const { map: locationsMap } = Locations.useMap();
+  const { mutate: update } = AT.useCreateOrUpdateMutation();
+
+  const handleEdit = item => {
+    const json = {
+      ...item,
+      remoteConfigurationId: configurationId,
+    };
+
+    return update(json);
+  };
+
+  return (
+    <EditableList
+      contentData={rows}
+      uniqueField="originalLocationId" // todo: file a bug in EditableList - ignored for choosing of onCreate, onUpdate
+      isEmptyMessage={<LoadingCentered />}
+      canCreate={false} // does not really help with "Create" button, just disables it
+      actionProps={{
+        create: () => ({ style: { display: 'none' } }), // to get rid of "Create" button
```

Comment:
Yeah, this smart component suddenly happened to be too smart )

Reply:
Yeah, I think a lot of the smart-components stuff was written sort of inside out, i.e. solve the specific problem and then make it generic, rather than cohesively designed to be generic in the first place. That's not meant to be a critique; just a statement of how we got here.